src/Explorer/Controls/Settings/SettingsSubComponents/MongoIndexingPolicy/MongoIndexingPolicyComponent.tsx

```diff
@@ -0,0 +1,406 @@
+
+interface MongoIndexDisplayProps {
+  definition: JSX.Element;
+  type: JSX.Element;
+  actionButton: JSX.Element;
+}
+
+export class MongoIndexingPolicyComponent extends React.Component<
+  MongoIndexingPolicyComponentProps,
+  MongoIndexingPolicyComponentState
+> {
+  private shouldCheckComponentIsDirty = true;
+  private currentTextFields: ITextField[] = [];
+  private initialIndexesColumns: IColumn[] = [
+    { key: "definition", name: "Definition", fieldName: "definition", minWidth: 100, maxWidth: 200, isResizable: true },
+    { key: "type", name: "Type", fieldName: "type", minWidth: 100, maxWidth: 200, isResizable: true },
+    {
+      key: "actionButton",
+      name: "Drop Index",
+      fieldName: "actionButton",
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true
+    }
+  ];
+
+  private indexesToBeDroppedColumns: IColumn[] = [
+    { key: "definition", name: "Definition", fieldName: "definition", minWidth: 100, maxWidth: 200, isResizable: true },
+    { key: "type", name: "Type", fieldName: "type", minWidth: 100, maxWidth: 200, isResizable: true },
+    {
+      key: "actionButton",
+      name: "Add index back",
+      fieldName: "actionButton",
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true
+    }
+  ];
+
+  constructor(props: MongoIndexingPolicyComponentProps) {
+    super(props);
+    this.state = {
+      indexesToBeAddedExpanded: true,
+      indexesToBeDroppedExpanded: true,
+      initialIndexesExpanded: true,
+      isRefreshingIndexTransformationProgress: false
+    };
+  }
+
+  componentDidUpdate(prevProps: MongoIndexingPolicyComponentProps): void {
+    if (this.props.indexesToAdd.length > prevProps.indexesToAdd.length) {
+      this.currentTextFields[prevProps.indexesToAdd.length]?.focus();
+    }
+    this.onComponentUpdate();
+  }
+
+  componentDidMount(): void {
+    this.onComponentUpdate();
+  }
+
+  private onComponentUpdate = (): void => {
+    if (!this.shouldCheckComponentIsDirty) {
+      this.shouldCheckComponentIsDirty = true;
+      return;
+    }
+    this.props.onMongoIndexingPolicySaveableChange(this.isMongoIndexingPolicySaveable());
+    this.props.onMongoIndexingPolicyDiscardableChange(this.isMongoIndexingPolicyDiscardable());
+    this.shouldCheckComponentIsDirty = false;
+  };
+
+  private toggleIndexesToBeDroppedExpanded = (): void => {
+    this.setState({ indexesToBeDroppedExpanded: !this.state.indexesToBeDroppedExpanded });
+  };
+
+  private toggleInitialIndexesExpanded = (): void => {
+    this.setState({ initialIndexesExpanded: !this.state.initialIndexesExpanded });
+  };
+
+  public isMongoIndexingPolicySaveable = (): boolean => {
+    const addErrorsExist = this.props.indexesToAdd.find(
+      (addMongoIndexProps: AddMongoIndexProps) => addMongoIndexProps.notification
+    );
+    return (this.props.indexesToAdd.length > 0 && !addErrorsExist) || this.props.indexesToDrop.length > 0;
+  };
+
+  public isMongoIndexingPolicyDiscardable = (): boolean => {
```

Comment:
`isMongoIndexingPolicyChangesDiscardable`  rather?

Reply:
I d rather keep it this way, otherwise the update function for this (now named onMongoIndexingPolicyDiscardableChange) will become onMongoIndexingPolicyChangeDiscardableChange, which is not great